### PR TITLE
fix: set environment variable for cuda

### PR DESCRIPTION
**Pull request by sebastian-weisshaar**
With the new version of transformers we cannot send the model to a device if we use quantization (https://github.com/huggingface/transformers/blob/66954ea25e342fd451c26ec1c295da0b8692086b/src/transformers/modeling_utils.py#L1897). To solve this we had to specify the accelerate version, instead of using the main branch on GitHub. 

This PR includes: https://github.com/jina-ai/jerboa/pull/102. 

THIS CODE DOES NOT RUN ON APPLE SILICON

### CODE REVIEW

File: jerboa/finetune.py [hunk 1]
```diff
@@ -166,9 +162,13 @@ def train(
 
     prompter = Prompter(prompt_template_name)
 
-    device_map = "auto"
+    # Device_map is not compatible with CPU training
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    device_map = "auto" if device == "cuda" else None
```
Comment (bwanglzu) [inline jerboa/finetune.py:167]:
```suggestion
    device_map = "auto" if device == 'cuda' else None
```

File: jerboa/finetune.py [hunk 2]
```diff
@@ -166,9 +162,13 @@ def train(
...
     world_size = int(os.environ.get("WORLD_SIZE", 1))
-    ddp = world_size != 1
+
+    # Only apply if we have more than one GPU, not if we have no GPU
+    ddp = world_size > 1
```
Comment (bwanglzu) [inline jerboa/finetune.py:171]: not important, i would rather implement a utilitity function called `is_distribuetd` to determine if it is DDP, an easy way is : 

```python
  if 'WORLD_SIZE' in os.environ:
      return int(os.environ['WORLD_SIZE']) > 1
```

**Review by bwanglzu — APPROVED** [review]
left minor comments

File: jerboa/finetune.py (hunk 2 above)
Comment (sebastian-weisshaar) [reply jerboa/finetune.py:171]: This returns a key error if we have no gpu

**Review by JohannesMessner — CHANGES_REQUESTED** [review]
Instead of pinning an older version of Transformers, is there a different/more proper way to handle this?
They probably block this in the new version for a reason, so seems like a bit of a code smell to keep using `to()` on quantized models.
Do they not have any guidance on what to do?

**Comment by sebastian-weisshaar** [comment]
> Instead of pinning an older version of Transformers, is there a different/more proper way to handle this?
> They probably block this in the new version for a reason, so seems like a bit of a code smell to keep using `to()` on quantized models.
> Do they not have any guidance on what to do?

We do not use `.to(device)`. Somewhere deep in transformers they call it when loading the model. Tried to workaround with the `device_map` but this did not stop transformers from calling `.to(device)`

**Review by JohannesMessner — APPROVED** [review]

**Pull request merged**